src/wolfsftp.c

```diff
@@ -3561,19 +3648,140 @@ int SFTP_GetAttributes(const char* fileName, WS_SFTP_FILEATRB* atr, byte link,
     return WS_SUCCESS;
 }
 
+#elif defined(FREESCALE_MQX)
+
+/* @TODO can be overriden by user for portability
+ * NOTE: if atr->flags is set to a value of 0 then no attributes are set.
+ * Fills out a WS_SFTP_FILEATRB structure
+ * returns WS_SUCCESS on success */
+int SFTP_GetAttributes(void* fs, const char* fileName, WS_SFTP_FILEATRB* atr,
+                       byte link, void* heap)
+{
+    int err, sz;
+    MQX_FILE_PTR mfs_ptr;
+    MFS_SEARCH_DATA search_data;
+    MFS_SEARCH_PARAM search;
+
+    if (fileName == NULL || atr == NULL) {
+        return WS_BAD_FILE_E;
+    }
+    mfs_ptr = (MQX_FILE_PTR)fs;
+    sz = (int)WSTRLEN(fileName);
+
+    /* handle case of '<drive>:/.' */
+    if (WSTRNCMP(fileName + sz - 3, ":/.", 3) == 0) {
+        atr->flags |= WOLFSSH_FILEATRB_PERM;
+        atr->per |= 0x4000;
+        return WS_SUCCESS;
+    }
+
+    search.ATTRIBUTE = MFS_SEARCH_ANY;
+    search.WILDCARD = (char*)fileName;
+    search.SEARCH_DATA_PTR = &search_data;
+
+    err = ioctl(mfs_ptr, IO_IOCTL_FIND_FIRST_FILE, (uint32_t*)&search);
```

Comment:
can we wrap this in WIOCTL?

Reply:
This may be challenging, as ioctl here accepts a MQX_FILE_PTR structure as the first argument, where when wolfSSH ssh.c calls WIOCTL for socket operations, it passes an "int fd" as the first argument.  We may need a separate abstraction for one of these use cases if we want to abstract this call out.